src/screens/PrivateChat/ContactsPicker/index.js

```diff
@@ -0,0 +1,259 @@
+import InvalidKeyModal from './InvalidKeyModal';
+import AssetsImage from '../../../global/AssetsImages';
+import { imageSource } from '../../../utils/profile';
+
+import { Chip, Selectize } from 'react-native-material-selectize';
+
+const DONE_BUTTON = 'DONE_BUTTON';
+const INVALID_MODAL_KEY = 'invalidKey';
+
+type Props = {
+  /**
+   * @desc React Native Navigation navigator object.
+   */
+  navigator: Navigator,
+  /**
+   * @desc List of all contacts
+   */
+  contacts: Array<*>,
+  /**
+   * @desc Function to get user profile
+   * @param {string} identityKey Identity key of user.
+   * @param {function} callback Callback
+   */
+  getProfile: (identityKey: string, callback: (profile: (ProfileType | null), error: (Error | null)) => void) => void,
+  /**
+   * @desc Function to add a new contact
+   * @param {string} identityKey Identity key of user.
+   * @param {function} callback Callback
+   */
+  addContact: (identityKey: string, callback: (error: Error | null) => void) => void,
+  /**
+   * @desc Function to initialize a new chat
+   * @param {ProfileType} profile Profile of the user
+   * @param {func} callback
+   */
+  createNewSession: (profile: ProfileType, callback: (result: Object) => void) => void,
+};
+
+type State = {
+  /**
+   * @desc Name of the modal to be shown
+   */
+  showModal: string,
+  /**
+   * @desc Flag whether loading is in progress.
+   */
+  loading: boolean,
+  /**
+   * @desc Error message for adding contact failure.
+   */
+  addContactError: string,
+};
+
+class ContactsPickerScreen extends NavigatorComponent<Props, State> {
+  static navigatorButtons = {
+    leftButtons: [],
+    rightButtons: [{
+      title: 'Done',
+      id: DONE_BUTTON,
+      buttonColor: Colors.navigationButtonColor,
+    }],
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: '',
+      loading: false,
+      addContactError: '',
+    };
+  }
+
+  selectize: Selectize;
+
+  onNavBarButtonPress(id) {
+    if (id === DONE_BUTTON) {
+      const selectedContacts = this.selectize.getSelectedItems().result;
+      if (selectedContacts.length) {
+        // @todo Create a new chat session from selected contacts
+      }
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.contacts.length == prevProps.contacts.length + 1) {
+      const newContact = this.props.contacts[this.props.contacts.length - 1];
+      this.selectize._selectItem(newContact.id)
+    }
+  }
+
+  getPublicKeyFromClipboard = async () => {
+    const pubKey = await Clipboard.getString();
+
+    return this.getUserProfile(pubKey);
+  };
+
+  getUserProfile = async publicKey => new Promise((res, rej) => {
+    this.props.getProfile(publicKey, (profile, error) => {
+      if (profile != null) {
+        res(profile);
+        return;
+      }
+
+      if (error != null) {
+        console.log(`[TEST] Profile fetch error: ${error.message}`);
+      }
+      rej(error);
+    });
+  });
+
+  addContact = async () => {
```

Comment:
Logic of that method is incorrect. It should do the following:
 - show loading
 - call addContact with public key from clipboard
 - hide loading
 - show error in case of any
 - select new contact if it's added correctly

There is no need to fetch profile, it will be done by `addContact` action. Also, there is no need to show INVALID_MODAL_KEY because it doesn't show actual error. Also, loading will be hidden before `addContact` callback is called, which is incorrect.